generate_instructions.py

```diff
@@ -116,6 +138,26 @@ def __str__(self):
     def mark_complete(self):
         return Configuration([], self.current_blocks + self.final_blocks)
 
+    def __eq__(self, other):
```

Comment:
Doesn't Python have a concept of a `set`? Could we just use that to hold the blocks and use the existing `Block.__eq__` method to assert the configurations are equal? If so, we could remove most of the body of this method.

Reply:
It does, you just write something like `set([1, 2, 3])`. However, checking if two sets are equal would check that both sets have `__eq__` elements. I don't think that two blocks should require that they appear the same to be `__eq__`, so I can't do `set(a) == set(b)` unless I swapped out the `__eq__` method at runtime which seems like a bad idea. Sets would still be reasonable, but I can't find a way to get a specific element out of the set to check that they look the same given that the are `__eq__`.

All that being said, looking at it now I think I could use a dictionary to make it faster and easier to read.

Reply:
It looks much nicer now